File: wushengran/scala_tutorial/src/main/scala/chapter03/Test01_Operator.scala
package chapter03

object Test01_Operator {
  def main(args: Array[String]): Unit = {
    var r: Double = 10.0 / 3
    println("r = " + r.formatted("%8.2f"))

    val s1 = "abc"
    val s2 = new String("abc")

    println(s1 == s2) // 内容是否相等，与 java发生了变化，相当于 == 实现了 equals
    println(s1.equals(s2)) // 内容是否相等
    println(s1.eq(s2)) // scala 判断地址的方法

    var b: Byte = 10
    //b += 1 // 不会默认做强转
    println(b)
  }
}

File: wushengran/scala_tutorial/src/main/java/Test01_Operator.java
public class Test01_Operator {
    public static void main(String[] args) {
        String s1 = "abc";
        String s2 = new String("abc");
        System.out.println(s1 == s2); // 比较地址
        System.out.println(s1.equals(s2)); // 比较内容

        byte b1 = 10;
        b1 += 1; // 默认做了强转
        System.out.println(b1);

        b1 = b1++; // (tmp), 比较混乱
        System.out.println(b1);
    }
}
